lib/postgres.go

```diff
@@ -0,0 +1,1414 @@
+	LikedPostHash   *BlockHash
+	IsUnlike        bool `pg:",use_zero"`
+}
+
+// PGMetadataCreatorCoin represents CreatorCoinMetadataa
+type PGMetadataCreatorCoin struct {
+	tableName struct{} `pg:"pg_metadata_creator_coins"`
+
+	TransactionHash             *BlockHash `pg:",pk"`
+	ProfilePublicKey            []byte
+	OperationType               CreatorCoinOperationType `pg:",use_zero"`
+	BitCloutToSellNanos         uint64                   `pg:",use_zero"`
+	CreatorCoinToSellNanos      uint64                   `pg:",use_zero"`
+	BitCloutToAddNanos          uint64                   `pg:",use_zero"`
+	MinBitCloutExpectedNanos    uint64                   `pg:",use_zero"`
+	MinCreatorCoinExpectedNanos uint64                   `pg:",use_zero"`
+}
+
+// PGMetadataCreatorCoinTransfer represents CreatorCoinTransferMetadataa
+type PGMetadataCreatorCoinTransfer struct {
+	tableName struct{} `pg:"pg_metadata_creator_coin_transfers"`
+
+	TransactionHash            *BlockHash `pg:",pk"`
+	ProfilePublicKey           []byte
+	CreatorCoinToTransferNanos uint64 `pg:",use_zero"`
+	ReceiverPublicKey          []byte
+}
+
+// PGMetadataSwapIdentity represents SwapIdentityMetadataa
+type PGMetadataSwapIdentity struct {
+	tableName struct{} `pg:"pg_metadata_swap_identities"`
+
+	TransactionHash *BlockHash `pg:",pk"`
+	FromPublicKey   []byte
+	ToPublicKey     []byte
+}
+
+type PGNotification struct {
+	tableName struct{} `pg:"pg_notifications"`
+
+	TransactionHash *BlockHash `pg:",pk"`
+	Mined           bool
+	ToUser          []byte
+	FromUser        []byte
+	OtherUser       []byte
+	Type            NotificationType
+	Amount          uint64
+	PostHash        *BlockHash
+	Timestamp       uint64
+}
+
+type NotificationType uint8
+
+const (
+	NotificationUnknown NotificationType = iota
+	NotificationSendClout
+	NotificationLike
+	NotificationFollow
+	NotificationCoinPurchase
+	NotificationCoinTransfer
+	NotificationCoinDiamond
+	NotificationPostMention
+	NotificationPostReply
+	NotificationPostReclout
+)
+
+type PGProfile struct {
+	tableName struct{} `pg:"pg_profiles"`
+
+	PKID                    *PKID `pg:",pk"`
+	PublicKey               *PublicKey
+	Username                string
+	Description             string
+	ProfilePic              []byte
+	CreatorBasisPoints      uint64
+	BitCloutLockedNanos     uint64
+	NumberOfHolders         uint64
+	CoinsInCirculationNanos uint64
+	CoinWatermarkNanos      uint64
+}
+
+type PGPost struct {
+	tableName struct{} `pg:"pg_posts"`
+
+	PostHash          *BlockHash `pg:",pk"`
+	PosterPublicKey   []byte
+	ParentPostHash    *BlockHash
+	Body              string
+	RecloutedPostHash *BlockHash
+	QuotedReclout     bool
+	Timestamp         uint64
+	Hidden            bool
+	LikeCount         uint64
+	RecloutCount      uint64
+	QuoteRecloutCount uint64
+	DiamondCount      uint64
+	CommentCount      uint64
+	Pinned            bool
+	ExtraData         map[string][]byte
+}
+
+func (post *PGPost) NewPostEntry() *PostEntry {
```

Comment:
Why isn't this defined as static like:
```
func NewPostEntry() *PostEntry {
```

Reply:
The `PGPost` needs to either be an argument or the method needs to be defined on `PGPost`. IMO, defining it on `PGPost` makes the most sense.